### PR TITLE
Fix zero-based error line and column numbers printing when compiling

### LINKED ISSUE #27043: Wrong line  and column numbers in compile errors in cli
**Description:**

**Steps to reproduce:**
Please see the below code
```ballerina
public function main() {
   string p = 1234;
}
````

The line numbers and column numbers are zero-based
```
Compiling source
        nn2.bal
ERROR [nn2.bal:(1:14,1:18)] incompatible types: expected 'string', found 'int'
error: compilation contains errors
```

**Affected Versions:**
SLP6

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**
@dulajdilshan

### PR DESCRIPTION
**Pull request by dulajdilshan**
## Purpose
$title

Fixes #27043 

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Review by hasithaa — APPROVED** [review]

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/task/CompileTask.java [hunk 1]
```diff
@@ -63,9 +66,21 @@ public void execute(Project project) {
...
+    private String convertDiagnosticToString(Diagnostic diagnostic) {
+        LineRange lineRange = diagnostic.location().lineRange();
+
+        LineRange oneBasedLineRange = LineRange.from(
+                lineRange.filePath(),
+                LinePosition.from(lineRange.startLine().line() + 1, lineRange.startLine().offset() + 1),
+                LinePosition.from(lineRange.endLine().line() + 1, lineRange.endLine().offset() + 1));
+
+        return diagnostic.diagnosticInfo().severity().toString() + " [" +
+                oneBasedLineRange.filePath() + ":" + oneBasedLineRange + "] " + diagnostic.message();
```
Comment (pubudu91) [inline cli/ballerina-cli/src/main/java/io/ballerina/cli/task/CompileTask.java:84]: Is this the best way to handle this? :thinking: My concern here is that the `toString()` of the Diagnostic impl is duplicated here. How about creating a copy of the whole Diagnostic instance? Maybe add a builder API to `Diagnostic` to make it easier to create copies of Diagnostic instances?

Comment (pubudu91) [reply cli/ballerina-cli/src/main/java/io/ballerina/cli/task/CompileTask.java:84]: We can go with this for now though, and maybe revisit later.

Comment (dulajdilshan) [reply cli/ballerina-cli/src/main/java/io/ballerina/cli/task/CompileTask.java:84]: IMO, creating a new diagnostic kinda going out of its definition, isn't it? Here, We only need to change the position and print right ?

Comment (pubudu91) [reply cli/ballerina-cli/src/main/java/io/ballerina/cli/task/CompileTask.java:84]: Yeah, converting the diagnostic to a 1-based position. My concern is this feels like a rather ad hoc solution and duplicates the `toString()` of the diagnostic implementation. My point was whether we should consider having a proper util/API for converting a given diagnostic to 1-based positions. 

But I guess we can go with this for now and revisit this later.

**Review by pubudu91 — APPROVED** [review]

**Review by SupunS — APPROVED** [review]

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/task/CompileTask.java (hunk 1 above)
Comment (dulajdilshan) [reply cli/ballerina-cli/src/main/java/io/ballerina/cli/task/CompileTask.java:84]: Cool

**Pull request merged**